components/suggest/src/store.rs

```diff
@@ -130,13 +134,15 @@ impl SuggestStore {
     }
 
     /// Ingests new suggestions from Remote Settings.
+    #[handle_error(Error)]
     pub fn ingest(&self, constraints: SuggestIngestionConstraints) -> SuggestApiResult<()> {
-        Ok(self.inner.ingest(constraints)?)
+        self.inner.ingest(constraints)
     }
 
     /// Removes all content from the database.
+    #[handle_error(Error)]
```

Comment:
The main drawback to the `error_support` crate is this macro, which is slightly magical since it not only converts errors but also potentially reports them.  Overall it hasn't been that bad to work with in app-services though.  On the plus side, it removes some boilerplate in the inner code.

Another option would be manually doing running the conversion.  I think that would mean adding something like `.map_err(::error_support::convert_log_report_error)` at the end of all returns.

Reply:
I like it! (And I love the documentation you added for it ✨)

The reduced boilerplate is great, and I think it actually calls out "this call is special, its errors get reported" better than a `.map_err(...)` at each return would.